.github/workflows/ui.yml

```diff
@@ -0,0 +1,32 @@
+name: UI checks
+
+on: [push]
+
+jobs:
+ ui_job:
+    runs-on: ubuntu-latest
+    name: UI integrity check
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.ref }}
+      - name: Clean yarn.lock private dependencies
+        run: |
+          echo "Checking for private yarn.lock dependencies"
+          if grep -E "nexus\.vanti\.co\.uk/repository/npm-public/" ui/yarn.lock; then
+            echo "Private URLs found in yarn.lock"
+            echo "Auto cleaning yarn.lock dependencies"
+            echo "YARN_LOCK_CLEANED=true" >> $GITHUB_ENV
+            sed -i -e 's/nexus\.vanti\.co\.uk\/repository\/npm-public/registry\.npmjs\.org/g' ui/yarn.lock
```

Comment:
It occurred to me that this change is being made without testing that it works before committing the change. It might be worth running a `yarn install --frozen-lockfile` after the replace to make sure that the changes haven't broken dependency resolution.

Reply:
Ahh good shout. I wasn't familiar with yarn. It looks like --frozen-lockfile is getting deprecated so I've used the recommended flags. Thanks!